### ISSUE TITLE
 Connection to server lost (MQTT::NotConnectedException)

### ISSUE THREAD

**Issue by rynr** [njh/ruby-em-mqtt#11]
I just tried the example code of the `README.md`.
Unfortunately, I get the following error (indenpendently of I run the sender or the reciever-part, using ruby 2.2.1 and em-mqtt 0.0.3):
```
/home/rjung/.rvm/gems/ruby-2.2.1/gems/em-mqtt-0.0.3/lib/em/mqtt/client_connection.rb:63:in `unbind': Connection to server lost (MQTT::NotConnectedException)
    from /home/rjung/.rvm/gems/ruby-2.2.1/gems/eventmachine-1.0.7/lib/eventmachine.rb:1457:in `event_callback'
    from /home/rjung/.rvm/gems/ruby-2.2.1/gems/eventmachine-1.0.7/lib/eventmachine.rb:194:in `release_machine'
    from /home/rjung/.rvm/gems/ruby-2.2.1/gems/eventmachine-1.0.7/lib/eventmachine.rb:194:in `ensure in run'
    from /home/rjung/.rvm/gems/ruby-2.2.1/gems/eventmachine-1.0.7/lib/eventmachine.rb:218:in `run'
```
As far as I took a look at the issue, `EventMachine::MQTT::ClientConnection` uses `EM::Connection`, it seems like the connection is already closed, before anything happens.
So this could also be an issue with the communication to `test.mosquitto.org`.

**Comment by njh** [comment]
Hello,

One of the problems I have had with EventMachine is that when something goes wrong (such as an exception getting thrown), then the stack trace doesn't reveal what actually went wrong.

Combined with MQTT not being very good at telling clients about problems - the typical behaviour is to just disconnect a client when there is a problem, means that it can be difficult to know what is wrong.

I think it is likely that there is something wrong with the configuration you are using to connect to test.mosquitto.org. Have you tried other clients, such as my other ruby mqtt client, or the mosquitto command-line client?

I have just tested the version of ruby-em-mqtt on github master (4132c3e39e69d73d0e91de1e9a86b4d973e3dd07) and can confirm that I am able to publish and subscribe to ```test.mosquitto.org```

**Comment by rynr** [comment]
I could connect using the `mosquitto_sub` and `mosquitto_pub` commands.

Switching to master changed the em-version to 1.0.8, and I could connect.
So I'm fine like this, perhaps you could build a new version.

**Comment by njh** [comment]
Ah, ok. Perhaps it is a bug I have already fixed then.

Yes. I really should make a release.

**Comment by njh** [comment]
I have just published version 0.0.4:
* https://rubygems.org/gems/em-mqtt
* https://github.com/njh/ruby-em-mqtt/tree/v0.0.4
